contracts/sweep/BaseTokenSweep.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+/**
+ * @title Abstract upgradeable contract that provides the ability to sweep erroneous transferred tokens to a defined beneficiary address
+ *
+ * @dev Access control implementation is required for many functions by design
+ */
+abstract contract BaseTokenSweep is UUPSUpgradeable {
+    address public beneficiary;
+
+    event BeneficiaryUpdated(address indexed beneficiary);
+
+    /**
+     * @notice Set the beneficiary of the token sweep
+     *
+     * @dev Needs access control implemented in the inheriting contract
+     * @param _beneficiary The address of the beneficiary
+     */
+    function setBeneficiary(address _beneficiary) external virtual {
```

Comment:
suggestion: keep this function `internal` and remove the `virtual`.

This is a function that must have permission guarding, providing an implementation removes the obligation for the concrete contract to do so. By making this function internal then it's up to the concrete contract to choose if and how they set the beneficiary.
Some use cases may not allow the setting at all, others may wrap it with another update (e.g. updating the treasury), others may already attach a meaning to beneficiary and want to call it something else i.e. via a different method name.